packages/pools/src/router/__tests__/routes.spec.ts

```diff
@@ -1,412 +1,1137 @@
-import { Int } from "@keplr-wallet/unit";
+import { Dec, Int } from "@keplr-wallet/unit";
 
+import { NotEnoughLiquidityError } from "../../errors";
+import { Route, RouteWithInAmount } from "../route";
+import { RoutablePool } from "../types";
 import {
   allPools,
   makeDefaultTestRouterParams,
+  makeRouterWithForceRoutes,
   makeStablePool,
   makeWeightedPool,
 } from "./pool";
 
 describe("OptimizedRoutes", () => {
-  describe("Simple: picks higher liquidity routes", () => {
-    test("picks higher liquidity pool", async () => {
-      const pools = [
-        makeWeightedPool({
-          id: "2",
-          firstPoolAsset: { amount: "1000" },
-          secondPoolAsset: { amount: "1000" },
-        }),
-        makeWeightedPool(),
-      ];
-
-      const router = makeDefaultTestRouterParams({
-        pools,
-        incentivizedPoolIds: [],
-        stakeCurrencyMinDenom: "ufoo",
+  describe("getOptimizedRoutesByTokenIn", () => {
+    describe("handles", () => {
```

Comment:
I believe we should use `describe` sparingly. Nested code tends to be more difficult to read and maintain. It seems to me that `OptimizedRoutes` and handles do not offer much information, so we could possibly remove these `describe` blocks. Perhaps we could incorporate 'handles' directly within `it`.

Reply:
Generally I agree in code, but in this scenario IMO it helps keep the types of tests organized given how many there are. It creates a nice tree structure that helps surface the more important tests first, and helps categorize the test ("handles" tests are for testing common error cases/defaults). This tree structure is especially powerful in the jest VSCode extension.

Reply:
<img width="269" alt="Screenshot 2023-05-15 at 12 57 48 PM" src="https://github.com/osmosis-labs/osmosis-frontend/assets/4606373/bbd85d0c-cbf0-48cd-bc4f-fdf331aaed36">

Reply:
IMO inlining handles like `handles no pools`, `handles invalid tokens`, and `handled not enough liquidity` makes it easier to read for me and also makes me avoid looking consistently up and down. Likewise, if tests fill the list, we would have to constantly scroll up. I guess we could also rely on memory, but I believe inlining it would make it less cumbersome. wdyt?